### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-generic.git`](https://github.com/cisagov/skeleton-generic.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/skeleton-tf-module.git skeleton-tf-module
    cd skeleton-tf-module
    git remote add skeleton https://github.com/cisagov/skeleton-generic.git
    git remote set-url --push skeleton no_push
    git switch develop
    git checkout -b lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the branch,
   `commit`, and `push` your changes:

    ```console
    git add .github/workflows/build.yml 
    git commit --message="Resolve Lineage conflicts."
    git push --force --set-upstream origin lineage/skeleton
    ```

1. Wait for all the automated tests to pass.

1. Check the "Everything is cool" checkbox below:

    - [x] ✌️ The conflicts in this pull request have been resolved.

1. Mark this draft pull request "Ready for review".

------------

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -29,37 +29,27 @@ jobs:
...
+        uses: actions/cache@v2
         with:
-          path: ${{ steps.go-cache.outputs.dir }}
-          key: "${{ runner.os }}-go"
+          path: |
+            ${{ env.PIP_CACHE_DIR }}
+            ${{ env.PRE_COMMIT_CACHE_DIR }}
+            ${{ env.CURL_CACHE_DIR }}
+            ${{ steps.go-cache.outputs.dir }}
+          key: "lint-${{ runner.os }}-\
```
Comment (jsf9k) [inline .github/workflows/build.yml:53]: `go` is not mentioned in the key.  Should it be?

**Review by jsf9k — CHANGES_REQUESTED** [review]
Please see my one question.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mcdonnnj) [reply .github/workflows/build.yml:53]: The previous key didn't use any versioning so I left it off. I can add it back if desired, I'd just need to get the version so the cache gets updated if we change the Go version being installed.

Comment (dav3r) [reply .github/workflows/build.yml:53]: If we are including the Python version, then we probably ought to include the Go version as well, right?

Comment (jsf9k) [reply .github/workflows/build.yml:53]: I think it makes sense to include it here, and we should have done it in cisagov/skeleton-packer#42 as well.  [cisagov/skeleton-packer](https://github.com/skeleton-packer) will need a new PR.

What say you, @felddy and @hillaryj?

Comment (jsf9k) [reply .github/workflows/build.yml:53]: We opted to add the go version to the cache key in cisagov/skeleton-packer#43, so I did it here as well in commit 8c2277e.  

While making that change I noticed that the `.terraform` directory was never added to the list of cached directories, so I went ahead and made that change in commit 5855358.

**Review by jsf9k — APPROVED** [review]

**Review by dav3r — APPROVED** [review]
👍👍

**Comment by mcdonnnj** [comment]
@jsf9k Regarding 5855358, I believe the `.terraform` directory was left out because when the Actions workflow initializes, it uses the `-upgrade=true` flag. Terraform does this lazily and just pulls the latest versions regardless of what is installed. So if you already have the latest version, it will still download that version again, which defeats the purpose of caching these things.

Do we still want to cache the `.terraform` directory given that?

**Comment by jsf9k** [comment]
> @jsf9k Regarding [5855358](https://github.com/cisagov/skeleton-tf-module/commit/58553584fb20dfd429a0f00c87ecbf2265c565ac), I believe the `.terraform` directory was left out because when the Actions workflow initializes, it uses the `-upgrade=true` flag. Terraform does this lazily and just pulls the latest versions regardless of what is installed. So if you already have the latest version, it will still download that version again, which defeats the purpose of caching these things.
> 
> Do we still want to cache the `.terraform` directory given that?

Ah, OK.  I do remember that now that you mention it.  I will remove that commit and instead add a comment to explain why that directory is not cached.

**Review by mcdonnnj — APPROVED** [review]
Beautiful ✨

**Review by dav3r — APPROVED** [review]
👍 👍 👍

**Pull request merged**